api/tests/mocha/migrations/remove-user-language.spec.js

```diff
@@ -0,0 +1,144 @@
+    chai.expect(migration.run).to.be.a('function');
+  });
+
+  it('should do nothing if there are no users', () => {
+    userQuery.resolves({ rows: [] });
+    return migration.run().then(() => {
+      chai.expect(userQuery.callCount).to.equal(1);
+      chai.expect(bulkDocs.callCount).to.equal(0);
+    });
+  });
+
+  it('should remove the language value from all the users that have it', () => {
+    const userLang0 = {
+      doc: {
+        _id: 'org.couchdb.user:lang0',
+        language: 'en'
+      }
+    };
+    const userLang1 = {
+      doc: {
+        _id: 'org.couchdb.user:lang1',
+        language: 'es'
+      }
+    };
+    const userLang3 = {
+      doc: {
+        _id: 'org.couchdb.user:lang2',
+        language: null
+      }
+    };
+    const userNoLang0 = {
+      doc: {
+        _id: 'org.couchdb.user:noLang0',
+      }
+    };
+    const userNoLang1 = { };
+
+    userQuery.onFirstCall().resolves({ rows: [ userLang0, userNoLang0, userLang1, userNoLang1 ] });
+    userQuery.onSecondCall().resolves({ rows: [ userLang3 ] });
+    userQuery.onThirdCall().resolves({ });
+    bulkDocs.resolves();
+
+    return migration.run().then(() => {
+      chai.expect(userQuery.callCount).to.equal(3);
+      chai.expect(userQuery.args[0]).to.deep.equal([docByType, expectedOptions]);
+      expectedOptions.skip = 100;
+      chai.expect(userQuery.args[1]).to.deep.equal([docByType, expectedOptions]);
+      expectedOptions.skip = 200;
+      chai.expect(userQuery.args[2]).to.deep.equal([docByType, expectedOptions]);
+      chai.expect(bulkDocs.callCount).to.equal(2);
+      chai.expect(bulkDocs.calledWithExactly([ { _id: 'org.couchdb.user:lang0' }, { _id: 'org.couchdb.user:lang1' } ]))
+        .to.be.true;
+      chai.expect(bulkDocs.calledWithExactly([ { _id: 'org.couchdb.user:lang2' } ])).to.be.true;
+    });
+  });
+
+  it('should do nothing if none of the users have a language value', () => {
+    const userNoLang0 = {
+      doc: {
+        _id: 'org.couchdb.user:noLang0',
+      }
+    };
+    const userNoLang1 = {
+      doc: {
+        _id: 'org.couchdb.user:noLang1',
+      }
+    };
+
+    userQuery.onFirstCall().resolves({ rows: [ userNoLang0, userNoLang1 ] });
+    userQuery.onSecondCall().resolves({ rows: [] });
+
+    return migration.run().then(() => {
+      chai.expect(userQuery.callCount).to.equal(2);
+      chai.expect(userQuery.calledWithExactly(docByType, expectedOptions)).to.be.true;
+      expectedOptions.skip = 100;
+      chai.expect(userQuery.calledWithExactly(docByType, expectedOptions)).to.be.true;
+      chai.expect(bulkDocs.callCount).to.equal(0);
+    });
+  });
+
+  it('should throw an error if one occurs when querying', () => {
+    const message = 'Some Error';
+    userQuery.rejects(message);
+
+    return migration.run()
+      .catch((error) => {
+        chai.expect(userQuery.callCount).to.equal(1);
+        chai.expect(userQuery.calledWithExactly(docByType, expectedOptions)).to.be.true;
```

Comment:
Please use `.args[..]` and `.to.deep.equal` everywhere instead of `calledWithExactly`.